### PR TITLE
Geometry Primitives - Polygon Lattice Points

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by brebenelmihnea** [comment]
image

File: content/5_Plat/Geo_Pri.mdx [hunk 1]
```diff
@@ -312,6 +317,78 @@ int main() {
...
+	Point operator-(const Point &other) { return {x - other.x, y - other.y}; }
+	friend istream &operator>>(istream &in, Point &p) {
+		int x, y;
+		in >> p.x >> p.y;
+		return in;
+	}
+};
+// EndCodeSnip
+
+int gcd(long long x, long long y) {
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:362]: p sure cpp has a builtin gcd function

File: content/5_Plat/Geo_Pri.mdx [hunk 2]
```diff
@@ -312,6 +317,78 @@ int main() {
...
+	if (!y) return x;
+	return gcd(y, x % y);
+}
+
+int main() {
+	int n;
+	cin >> n;
+	vector<Point> points(n);
+	for (auto &point : points) { cin >> point; }
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:371]:
```suggestion
	for (Point &point : points) { cin >> point; }
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
W

File: content/5_Plat/Geo_Pri.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx]: All right I used __gcd

File: content/5_Plat/Geo_Pri.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx]: Done

File: content/5_Plat/Geo_Pri.mdx [hunk 3]
```diff
@@ -379,7 +374,7 @@ int main() {
 	long long boundary_points = 0;
 	for (int i = 0; i < n; i++) {
 		Point diff = points[i + 1] - points[i];
-		int g = gcd(abs(diff.x), abs(diff.y));
+		int g = __gcd(abs(diff.x), abs(diff.y));
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:377]: really sorry but could you use [this gcd](https://en.cppreference.com/w/cpp/numeric/gcd) instead?

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx]: Done

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
solid!

**Pull request merged**